configs/internal_flash_dual_bank_no_rot.json

```diff
@@ -0,0 +1,110 @@
+                    "|        Active App        |",
+                    "|                          |",
+                    "|                          |",
+                    "|                          |",
+                    "+--------------------------+ <-+ mbed-bootloader.application-start-address",
+                    "|Active App Metadata Header|",
+                    "+--------------------------+ <-+ update-client.application-details",
+                    "|                          |",
+                    "|        Bootloader        |",
+                    "|                          |",
+                    "+--------------------------+ <-+ 0",
+                    "Please note that because the firmware candidate is stored on internal flash, the",
+                    "bootloader does not require a rot to operate. Hence the KVSTORE region is there for",
+                    "the benefit of Pelion Device Management Client only"
+            ]
+        },
+        "kvstore-size": {
+            "help": "The size of the area of flash reserved for kvstore, it neeeds to be even number of flash sectors",
+            "macro_name": "KVSTORE_SIZE",
+            "required": true
+        }
+    },
+    "macros": [
+        "MBED_CLOUD_CLIENT_UPDATE_STORAGE=ARM_UCP_FLASHIAP",
+        "MBED_BOOTLOADER_ACTIVE_HEADER_REGION_SIZE=1024",
+        "MBED_BOOTLOADER_FLASH_BANK_SIZE=MBED_ROM_SIZE/2"
+    ],
+    "target_overrides": {
+        "*": {
+            "target.console-uart"                      : false,
+            "target.c_lib"                             : "small",
+            "platform.default-serial-baud-rate"        : 115200,
+            "update-client.firmware-header-version"    : "2",
+            "mbed-bootloader.use-kvstore-rot"          : 0,
+            "mbed-bootloader.bootloader-size"          : "APPLICATION_SIZE",
+            "target.restrict_size"                     : "0x8000",
+            "kvstore-size"                             : "2*16*1024",
+            "update-client.application-details"        : "(MBED_ROM_START + MBED_BOOTLOADER_SIZE)",
+            "mbed-bootloader.application-start-address": "(MBED_CONF_UPDATE_CLIENT_APPLICATION_DETAILS + MBED_BOOTLOADER_ACTIVE_HEADER_REGION_SIZE)",
+            "mbed-bootloader.max-application-size"     : "(MBED_ROM_START + MBED_BOOTLOADER_FLASH_BANK_SIZE - MBED_CONF_MBED_BOOTLOADER_APPLICATION_START_ADDRESS)",
+            "update-client.storage-address"            : "(MBED_ROM_START + MBED_BOOTLOADER_FLASH_BANK_SIZE + KVSTORE_SIZE)",
+            "update-client.storage-size"               : "(MBED_BOOTLOADER_FLASH_BANK_SIZE - KVSTORE_SIZE)",
+            "update-client.storage-locations"          : 1,
+            "target.extra_labels_remove"               : [ "PSA" ],
+            "update-client.storage-page"               : 1
+        },
+        "DISCO_F769NI": {
+            "target.flash_dual_bank"                   : 1,
```

Comment:
considering the file name, this config could be on target definitions. Similar thing may be needed for other F7-series boards also.

Reply:
`target.flash_dual_bank` is specific only for some F76X-boards. At the moment I would only keep it in the device-specific override.

Reply:
yes
https://os.mbed.com/teams/ST/wiki/How-to-enable-flash-dual-bank